optimum/exporters/onnx/base.py

```diff
@@ -558,3 +558,93 @@ def get_decoder_onnx_config(
             f"{config.model_type} decoder export is not supported yet. ",
             f"If you want to support {config.model_type} please propose a PR or open up an issue.",
         )
+
+
+class OnnxConfigWithLoss(OnnxConfig, ABC):
+    """
+    Wrapper for the children classes of `optimum.exporters.onnx.OnnxConfig` to export the model through the ONNX format
+    with loss in outputs and labels in the inputs. For seq-to-seq models, labels will be appended to the inputs of
+    decoders.
+    """
+
+    _tasks_to_extra_inputs = {
+        "default": {"labels": {0: "batch_size"}},
+        "masked-lm": {"labels": {0: "batch_size", 1: "sequence_length"}},
+        "causal-lm": {"labels": {0: "batch_size", 1: "sequence_length"}},
+        "seq2seq-lm": {"labels": {0: "batch_size", 1: "sequence_length"}},
+        "sequence-classification": {"labels": {0: "batch_size"}},
+        "token-classification": {"labels": {0: "batch_size", 1: "sequence_length"}},
+        "multiple-choice": {"labels": {0: "batch_size"}},
+        "question-answering": {
+            "start_positions": {0: "batch_size"},
+            "end_positions": {0: "batch_size"},
+        },
+        "image-classification": {"labels": {0: "batch_size"}},
+        "seq2seq-lm": {"labels": {0: "batch_size", 1: "sequence_length"}},
+    }
+    _tasks_to_extra_outputs = {
+        "default": OrderedDict({"loss": {}}),
+    }
+
+    DUMMY_INPUT_GENERATOR_CLASSES = (DummyTrainingLabelsInputGenerator,)
+
+    def __init__(self, config: OnnxConfig):
+        self.__dict__ = copy.deepcopy(config.__dict__)
+        self._onnx_config = config
```

Comment:
I do not think this is nice for the `OnnxConfigWithLoss` to be both an `OnnxConfig`, with the same state as `config` and to have it as an attribute that is being used in methods. If it has the same state, why should we need to store `config`?

For instance the two properties `torch_to_onnx_input_map` and `values_override` are not needed no?
Same thing for `generate_dummy_inputs_for_validation`?

Maybe I am missing something but wanted to raise the question!

Reply:
Yes you are right.  The old `OnnxConfigWithLoss` also has both the state and config copied. But I guess we don't need both. I could remove the copying of the state, this will also remove the new condition in `normalised_config.py` for deep copy.

Answering why `torch_to_onnx_input_map` and `values_override`  are needed - The `OnnxConfigWithLoss` would be a wrapper for every model config, be it Bert, bart, t5 etc. These methods could have different implementations for each model in their config, plus there could be additional methods as well. So it would be better to return the `_onnx_config` method output when they are called.

Same case is for `generate_dummy_inputs_for_validation`. This method also uses the `_onnx_config` to return the old inputs and then appends the extra inputs.

Reply:
Ok I understand now! Then yes, I think `OnnxConfigWithLoss` should not be an `OnnxConfig` and just a wrapper around it.